src/dowel/csv_output.py

```diff
@@ -42,12 +40,23 @@ def record(self, data, prefix=''):
                 self._writer.writeheader()
 
             if to_csv.keys() != self._fieldnames:
-                self._warn('Inconsistent TabularInput keys detected. '
-                           'CsvOutput keys: {}. '
-                           'TabularInput keys: {}. '
-                           'Did you change key sets after your first '
-                           'logger.log(TabularInput)?'.format(
-                               set(self._fieldnames), set(to_csv.keys())))
+                              
+             #save the old file logs in a variable
+                with open(self._log_file.name) as log_in_csv:
+                    old_data = csv.DictReader(log_in_csv)
+                    #update the fieldnames dictionary with additional keys
+                    self._fieldnames.update(set(to_csv.keys()))
+                    #update the writer object with new fieldnames
+                    self._writer = csv.DictWriter(self._log_file,fieldnames=self._fieldnames,extrasaction='raise')
```

Comment:
What's the point of constructing the writer here, then constructing it again two lines later?

Reply:
Fixed it, that was to update the headers but now its done in better way.